gtsfm/densify/patchmatchnet_data.py

```diff
@@ -0,0 +1,273 @@
+            from a common track's coordinates to each camera center, which is mentioned in "View Selection" paragraphs
+            in Yao's paper https://arxiv.org/abs/1804.02505.
+            3. For every image as the reference image, calculate the depth range.
+
+        Returns:
+            pairs: 2d array of shape (num_images, num_views-1). Each row_id indicates the index of reference view
+                in self.keys, with (num_views-1) values indicating the indices of source views in self.keys
+            depth_ranges: 2d array of shape (num_images, 2). Each row_id indicates the index of reference view
+                in self.keys, with 2 values indicating [min_depth, max_depth]
+        """
+        num_tracks = self._sfm_result.number_tracks()
+
+        # Initialize the pairwise scores between the same views as negative infinity
+        pair_scores = np.zeros((self._num_images_cal, self._num_images_cal))
+        np.fill_diagonal(pair_scores, -np.inf)
+
+        # Initialize empty lists to collect all possible depths for each view
+        depths = defaultdict(list)
+
+        for j in range(num_tracks):
+            track = self._sfm_result.get_track(j)
+            num_measurements = track.number_measurements()
+            measurements = [track.measurement(k) for k in range(num_measurements)]
+            w_x = track.point3()
+            for k1 in range(num_measurements):
+                i_a = measurements[k1][0]
+                # Check if i_a is a valid image with estimated camera pose, then get its id for patchmatchnet
+                i_pm_a = self._camera_idx_to_patchmatchnet_idx.get(i_a, -1)
+
+                # Calculate track_i's depth in the camera pose
+                z_a = self._sfm_result.get_camera(i_a).pose().transformTo(w_x)[-1]
+                # Update image i's depth list only when i in a valid camera
+                depths[i_pm_a].append(z_a)
+
+                for k2 in range(k1 + 1, num_measurements):
+                    i_b = measurements[k2][0]
+
+                    # Check if i_b is a valid image with estimated camera pose, then get its id for patchmatchnet
+                    i_pm_b = self._camera_idx_to_patchmatchnet_idx.get(i_b, -1)
+
+                    # Calculate the pairwise gaussian score only if i_a and i_b are both images with estimated poses
+                    if i_pm_a < 0 or i_pm_b < 0:
+                        continue
+
+                    # If both cameras are valid cameras,
+                    #   calculate score for track_i in the pair views (cam_a, cam_b)
+                    score_a_b = mvs_utils.piecewise_gaussian(
+                        xPa=self._camera_centers[i_a] - w_x, xPb=self._camera_centers[i_b] - w_x
+                    )
+                    # Sum up pair scores for each track_i
+                    pair_scores[i_pm_b, i_pm_a] += score_a_b
+                    pair_scores[i_pm_a, i_pm_b] += score_a_b
+
```

Comment:
same renaming comments here: i2, pm_i2, score_i1_i2

Reply:
agreed on this

Reply:
thanks, resolved